File: Eje_Semana_6/Python/SistemaProductos.py
class Productos:
    def __init__(self, nombre, precio, categoria):
        self.nombre = nombre
        self.precio = precio
        self.categoria = categoria

    def mostrar_detalle(self):
        return f"Nombre: {self.nombre}\nPrecio: {self.precio}\nCategoría: {self.categoria}"

class Electronico(Productos):
    def __init__(self, nombre, precio, categoria, marca, modelo):
        super().__init__(nombre, precio, categoria)
        self.marca = marca
        self.modelo = modelo

    def mostrar_detalle(self):
        padre = super().mostrar_detalle()
        return f"{padre}\nMarca: {self.marca}\nModelo: {self.modelo}"

class Alimenticio(Productos):
    def __init__(self, nombre, precio, categoria, fecha_vencimiento):
        super().__init__(nombre, precio, categoria)
        self.fecha_vencimiento = fecha_vencimiento

    def mostrar_detalle(self):
        padre = super().mostrar_detalle()
        return f"{padre}\nFecha de Vencimiento: {self.fecha_vencimiento}"

class Vestimenta(Productos):
    def __init__(self, nombre, precio, categoria, talla, color):
        super().__init__(nombre, precio, categoria)
        self.talla = talla
        self.color = color

    def mostrar_detalle(self):
        padre = super().mostrar_detalle()
        return f"{padre}\nTalla: {self.talla}\nColor: {self.color}"


electronico = Electronico("Laptop", 1000, "Electrónico", "Dell", "XPS 15")
alimenticio = Alimenticio("Leche", 2.5, "Alimenticio", "2023-10-15")
vestimenta = Vestimenta("Camiseta", 15, "Vestimenta", "M", "Azul")


print("Detalles del producto electrónico:")
print(electronico.mostrar_detalle())

print("\nDetalles del producto alimenticio:")
print(alimenticio.mostrar_detalle())

print("\nDetalles del producto de vestimenta:")
print(vestimenta.mostrar_detalle())

File: Eje_Semana_6/Java/SistemaProductos.java
public class SistemaProductos {

    public static class Productos {
        private String nombre;
        private double precio;
        private String categoria;

        public Productos(String nombre, double precio, String categoria) {
            this.nombre = nombre;
            this.precio = precio;
            this.categoria = categoria;
        }

        public String mostrarDetalle() {
            return "Nombre: " + nombre + "\nPrecio: " + precio + "\nCategoría: " + categoria;
        }
    }

    public static class Electronico extends Productos {
        private String marca;
        private String modelo;

        public Electronico(String nombre, double precio, String categoria, String marca, String modelo) {
            super(nombre, precio, categoria);
            this.marca = marca;
            this.modelo = modelo;
        }

        public String mostrarDetalle() {
            String padre = super.mostrarDetalle();
            return padre + "\nMarca: " + marca + "\nModelo: " + modelo;
        }
    }

    public static class Alimenticio extends Productos {
        private String fechaVencimiento;

        public Alimenticio(String nombre, double precio, String categoria, String fechaVencimiento) {
            super(nombre, precio, categoria);
            this.fechaVencimiento = fechaVencimiento;
        }


        public String mostrarDetalle() {
            String padre = super.mostrarDetalle();
            return padre + "\nFecha de Vencimiento: " + fechaVencimiento;
        }
    }

    public static class Vestimenta extends Productos {
        private String talla;
        private String color;

        public Vestimenta(String nombre, double precio, String categoria, String talla, String color) {
            super(nombre, precio, categoria);
            this.talla = talla;
            this.color = color;
        }


        public String mostrarDetalle() {
            String padre = super.mostrarDetalle();
            return padre + "\nTalla: " + talla + "\nColor: " + color;
        }
    }

    public static void main(String[] args) {
        Electronico electronico = new Electronico("Laptop", 1000, "Electrónico", "Dell", "XPS 15");
        Alimenticio alimenticio = new Alimenticio("Leche", 2.5, "Alimenticio", "2023-10-15");
        Vestimenta vestimenta = new Vestimenta("Camiseta", 15, "Vestimenta", "M", "Azul");

        System.out.println("Detalles del producto electrónico:");
        System.out.println(electronico.mostrarDetalle());

        System.out.println("\nDetalles del producto alimenticio:");
        System.out.println(alimenticio.mostrarDetalle());

        System.out.println("\nDetalles del producto de vestimenta:");
        System.out.println(vestimenta.mostrarDetalle());
    }
}
